api/control/v1alpha1/types.go

```diff
@@ -329,8 +342,9 @@ type ClusterScopeDownsyncClause struct {
 	// +optional
 	CreateOnly bool `json:"createOnly,omitempty"`
 
-	// `statusCollectors` is a list of StatusCollectors name references that are applied to the object.
-	StatusCollectors []string `json:"statusCollectors,omitempty"`
+	// `statusCollection` holds the rules of status collection for the object.
+	// +optional
+	StatusCollection *StatusCollection `json:"statusCollection,omitempty"`
 }
 
```

Comment:
I think that the types.go file should have a declaration of the data structure that the expressions are evaluated against. I see that this PR currently documents that in the user guide, but: (A) the description there is not as precise as a data structure, and programmers need precision, and (B) types.go is where programmers go for the real API definition.

It could look something like the following.

```go
// ExpressionContext defines what an Expression can reference regarding the workload object.
type ExpressionContext struct {
	// `inventory` holds the inventory record for the workload object
	Inventory InventoryRecord `json:"inventory"`

	// `obj` holds a copy of the workload object as read from the WDS
	Obj map[string]any `json:"obj"`

	// `returned` holds the fragment of the workload object that was returned to the core from the WEC
	Returned ReturnedState `json:"returned"`

	// `propagation` holds data about the current state of the work on propagating
	// the object's state from WDS to WEC and from WEC to WDS.
	Propagation PropagatationData `json:"propagatation"`
}

// InventoryRecord is what appears in the inventory for a given WEC.
type InventoryRecord struct {
	// the name of the WEC
	Name string `json:"name"`
}

type ReturnedState struct {
	Status map[string]any `json:"status"`
}

type PropagatationData struct {
	// `lastReturnedUpdateTimestamp` is the time of the last update to any
	// of the returned object state in the core.
	// Before the first such update, this is the zero value of `time.Time`.
	LastReturnedUpdateTimestamp time.Time `json:"lastReturnedUpdateTimestamp"`
}
```